lib/auth/auth.go

```diff
@@ -3081,10 +3100,11 @@ func generateCert(ctx context.Context, a *Server, req certRequest, caType types.
 	}
 
 	var signedTLSCert []byte
-	notAfter := a.clock.Now().UTC().Add(sessionTTL)
-	// generate TLS certificate if the signing CA isn't OpenSSH CA, as
-	// OpenSSH CA doesn't have any TLS keypairs
-	if caType != types.OpenSSHCA {
+	if req.tlsPublicKey != nil {
```

Comment:
`GenerateOpenSSHCert` now doesn't pass `tlsPublicKey`, so we don't need to check the CA type